mcstatus/server.py

```diff
@@ -51,14 +66,24 @@ def port(self) -> int:
         return self.address.port
 
     @classmethod
+    @abstractmethod
     def lookup(cls, address: str, timeout: float = 3) -> Self:
-        """Mimics minecraft's server address field.
+        """Lookup a minecraft server. This can have additional logic in children classes."""
 
-        :param str address: The address of the Minecraft server, like `example.com:19132`
-        :param float timeout: The timeout in seconds before failing to connect.
-        """
-        addr = Address.parse_address(address, default_port=None)
-        return cls(addr.host, addr.port, timeout=timeout)
+    @classmethod
+    @abstractmethod
+    async def async_lookup(cls, address: str, timeout: float = 3) -> Self:
+        """Async lookup a minecraft server. This can have additional logic in children classes."""
```

Comment:
Async lookups don't make sense for Bedrock, since there's no actual lookup being performed, it's handled purely with `Address.parse_address`, which simply takes the string IP:Port and obtains the individual parts (IP and port).

Async behavior only makes sense with Java lookups, which require an additional SRV DNS record check, that does need an asynchronous version. This means that the base class should only include the synchronous lookup, since only that is shared across both servers, and therefore only sync version should be in the base class

Reply:
This was made for the same interface between `JavaServer` and `BedrockServer`, easier will be create proxy-method than check `isinstance` in client code. So client code will be like this:
```py
server = await object_by_someone_returned_and_type_mc_server.async_lookup(ip)
```
Instead of:
```py
obj = object_by_someone_returned_and_type_mc_server

if isinstance(obj, JavaServer):
    server = await obj.async_lookup(ip)
elif isinstance(obj, BedrockServer):
    server = obj.lookup(ip)
```

Reply:
Yeah, I understand the intention behind it, but let's schedule this for the cross-platform PR, and we can decide whether it's a good idea there, but it certainly doesn't belong to this PR, which is only supposed to introduce the response classes

Reply:
Deleted in 4ecf8832e0895740cac091345225ac45603e6ddf.